p2p/p2p.go

```diff
@@ -59,6 +59,14 @@ type HostConfig struct {
 	DHTOpts      []dht.Option
 }
 
+type IDService interface {
```

Comment:
What is the reason to use your own interface declaration, but not already existed struct with same public funcs?

Reply:
Hm, no particular reasons, in next libp2p release this struct became private, so I thought to use interface in advance. Will remove this interface after upgrade.